### PR TITLE
Testing (RxJS/NgRx) observables (async operations) and logic stored in ng"Effects"

### PR DESCRIPTION
**Pull request by ammarhaiderbjss**
## Description and relevant Jira numbers

Full confluence write-up here - https://wiki.i-env.net/pages/viewpage.action?pageId=34594088

## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop
- [ ] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [x] PR link added to JIRA
- [x] Reviewers selected in Github
- [ ] Any new reusable component/widget added to component library on Confluence

- [ ] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

### CODE REVIEW

File: src/pages/journal/__tests__/journal.effects.spec.ts [hunk 1]
```diff
@@ -30,38 +36,136 @@ export class TestActions extends Actions {
...
+    // ACT
+    actions$.next(new journalActions.SelectPreviousDay());
+    // ASSERT
+    effects.selectPreviousDayEffect$.subscribe((result) => {
+      expect(store$.dispatch).toHaveBeenCalledWith(new journalActions.JournalNavigateDay(selectedDate));
+      expect(result).toEqual(new journalActions.SetSelectedDate(selectedDate));
+      done();
+    });
+
+  });
```
Comment (buzzub) [inline src/pages/journal/__tests__/journal.effects.spec.ts]: This looks really neat. I like how each test is commented with arrange, act and assert. Very simple to follow.

File: package-lock.json [hunk 2]
```diff
@@ -14939,7 +14939,7 @@
     },
     "pegjs": {
       "version": "0.10.0",
-      "resolved": "https://registry.npmjs.org/pegjs/-/pegjs-0.10.0.tgz",
+      "resolved": "http://registry.npmjs.org/pegjs/-/pegjs-0.10.0.tgz",
```
Comment (johnmorleybjss) [inline package-lock.json]: Slightly unrelated, but I wonder why these URL schemes change. I saw a PR changing `http` to `https` yesterday.

Comment (ammarhaiderbjss) [reply package-lock.json]: I've no idea, good spot!

**Comment by ammarhaiderbjss** [comment]
> Great that we've finally got some tests around Effects, however, this isn't quite the outcome I had expected from this story.
> 
> I was expecting to see some marble tests allowing us to try out various different permutations of emissions from source observables to ensure that we have covered all possible eventualities that could occur at runtime. Did you do any investigation into marble testing as mentioned on the ticket? It feels like the style of tests you've written here wouldn't give us quite the same level of assurance as marble tests would. Happy to merge notwithstanding these queries.

Yes, marble testing was trialled at length but to no avail :( this is all documented in the write up - https://wiki.i-env.net/pages/viewpage.action?pageId=34594088

**Pull request merged**